### ISSUE TITLE
Fix S1104 FP: Should be ignored on classes and structs marked as [Serializable]

### ISSUE THREAD

**Issue by Dukobpa3** [SonarSource/sonar-dotnet#8504]
### Description

S1104 FP: Should be ignored on classes and structs marked as [Serializable]

### Repro steps

This kind of classes/structs always have an issue on each field.

```csharp
        [Serializable]
        public class ConfigOptionDto
        {
            public string type;

            public string key;

            public string value;
        }

```

### Expected behavior

If object have `[Serializable]` attribute then 99.9% it will have a lot of public fields, so this rule is not relevant in this case

### Actual behavior

This kind of classes/structs always have an issue on each field.

### Known workarounds
Something like this. But in this case we really don't need it since all compilers already understand `[Serializable]` objects and will not strip it, an even Rider/Resharper will not highlite this issue etc...

```
            [UsedImplicitly]
            public string type;

            [UsedImplicitly]
            public string key;

            [UsedImplicitly]
            public string value;
```

### Related information

* Visual Studio version – 2022
* MSBuild / dotnet version – latest from VS 2022
* SonarScanner for .NET version (if used) – 6.xx
* Operating System – Windows 10

**Comment by Tim-Pohlmann** [comment]
Hello @Dukobpa3, thanks for the report.
For clarification: Using properties instead of fields would be another workaround, correct? Is there a particular reason why you prefer fields over properties for serializable classes?

**Comment by Dukobpa3** [comment]
> Using properties instead of fields would be another workaround, correct?

Actually not because many serializers can't work with properties.

**Comment by Tim-Pohlmann** [comment]
I don't think making the fields private is a solution in most cases, so I clarified this to be an FP for S1104 but not S1144.

**Comment by Tim-Pohlmann** [comment]
For transparency: I updated the code sample to use a public class, since the rule doesn't raise for private ones.

**Comment by Dukobpa3** [comment]
S1144 I think relevant also

Sometimes I should add the fields just to copy serializing source structure, but not all of them will be used after that

Especially when working with some 3-d parties.
Or in case implementing network protocols with sending serialized data
I should implement some protocol DTO but not each field will be used after that

**Comment by Tim-Pohlmann** [comment]
@Dukobpa3 You are making some great points. I have created another issue. Thanks!
